In [60]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt

https://www.kaggle.com/datasets/mnassrib/jena-climate

In [61]:
df = pd.read_csv("../data/raw/jena_climate_2009_2016.csv")

## a

In [62]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 420551 entries, 0 to 420550
Data columns (total 15 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   Date Time        420551 non-null  object 
 1   p (mbar)         420551 non-null  float64
 2   T (degC)         420551 non-null  float64
 3   Tpot (K)         420551 non-null  float64
 4   Tdew (degC)      420551 non-null  float64
 5   rh (%)           420551 non-null  float64
 6   VPmax (mbar)     420551 non-null  float64
 7   VPact (mbar)     420551 non-null  float64
 8   VPdef (mbar)     420551 non-null  float64
 9   sh (g/kg)        420551 non-null  float64
 10  H2OC (mmol/mol)  420551 non-null  float64
 11  rho (g/m**3)     420551 non-null  float64
 12  wv (m/s)         420551 non-null  float64
 13  max. wv (m/s)    420551 non-null  float64
 14  wd (deg)         420551 non-null  float64
dtypes: float64(14), object(1)
memory usage: 48.1+ MB


In [63]:
df.head()

Date Time  p (mbar)  T (degC)  Tpot (K)  Tdew (degC)  rh (%)  \
0  01.01.2009 00:10:00    996.52     -8.02    265.40        -8.90    93.3   
1  01.01.2009 00:20:00    996.57     -8.41    265.01        -9.28    93.4   
2  01.01.2009 00:30:00    996.53     -8.51    264.91        -9.31    93.9   
3  01.01.2009 00:40:00    996.51     -8.31    265.12        -9.07    94.2   
4  01.01.2009 00:50:00    996.51     -8.27    265.15        -9.04    94.1   

   VPmax (mbar)  VPact (mbar)  VPdef (mbar)  sh (g/kg)  H2OC (mmol/mol)  \
0          3.33          3.11          0.22       1.94             3.12   
1          3.23          3.02          0.21       1.89             3.03   
2          3.21          3.01          0.20       1.88             3.02   
3          3.26          3.07          0.19       1.92             3.08   
4          3.27          3.08          0.19       1.92             3.09   

   rho (g/m**3)  wv (m/s)  max. wv (m/s)  wd (deg)  
0       1307.75      1.03           1.75     152.3  
1       1309.80      0.72           1.50     136.1  
2       1310.24      0.19           0.63     171.6  
3       1309.19      0.34           0.50     198.0  
4       1309.00      0.32           0.63     214.3

In [69]:
df.describe()

Date Time       p (mbar)       T (degC)  \
count                         420551  420551.000000  420551.000000   
mean   2012-12-30 06:26:51.498725120     989.212776       9.450147   
min              2009-01-01 00:10:00     913.600000     -23.010000   
25%              2010-12-31 03:25:00     984.200000       3.360000   
50%              2012-12-30 06:20:00     989.580000       9.420000   
75%              2014-12-29 18:55:00     994.720000      15.470000   
max              2017-01-01 00:00:00    1015.350000      37.280000   
std                              NaN       8.358481       8.423365   

            Tpot (K)    Tdew (degC)         rh (%)   VPmax (mbar)  \
count  420551.000000  420551.000000  420551.000000  420551.000000   
mean      283.492743       4.955854      76.008259      13.576251   
min       250.600000     -25.010000      12.950000       0.950000   
25%       277.430000       0.240000      65.210000       7.780000   
50%       283.470000       5.220000      79.300000      11.820000   
75%       289.530000      10.070000      89.400000      17.600000   
max       311.340000      23.110000     100.000000      63.770000   
std         8.504471       6.730674      16.476175       7.739020   

        VPact (mbar)   VPdef (mbar)      sh (g/kg)  H2OC (mmol/mol)  \
count  420551.000000  420551.000000  420551.000000    420551.000000   
mean        9.533756       4.042412       6.022408         9.640223   
min         0.790000       0.000000       0.500000         0.800000   
25%         6.210000       0.870000       3.920000         6.290000   
50%         8.860000       2.190000       5.590000         8.960000   
75%        12.350000       5.300000       7.800000        12.490000   
max        28.320000      46.010000      18.130000        28.820000   
std         4.184164       4.896851       2.656139         4.235395   

        rho (g/m**3)       wv (m/s)  max. wv (m/s)       wd (deg)  
count  420551.000000  420551.000000  420551.000000  420551.000000  
mean     1216.062748       1.702224       3.056555     174.743738  
min      1059.450000   -9999.000000   -9999.000000       0.000000  
25%      1187.490000       0.990000       1.760000     124.900000  
50%      1213.790000       1.760000       2.960000     198.100000  
75%      1242.770000       2.860000       4.740000     234.100000  
max      1393.540000      28.490000      23.500000     360.000000  
std        39.975208      65.446714      69.016932      86.681693

both wv (m/s) and max. wv (m/s) column contain error values (-9999)

In [65]:
print(f'number of columns contain error value: {len(df[df["wv (m/s)"] < 0])}')
df[df["wv (m/s)"] < 0].head()

number of columns contain error value: 18


Date Time  p (mbar)  T (degC)  Tpot (K)  Tdew (degC)  \
343578  13.07.2015 09:10:00    990.49     16.85    290.80        12.78   
343579  13.07.2015 09:20:00    990.32     17.06    291.03        12.86   
343580  13.07.2015 09:30:00    990.24     17.43    291.41        12.75   
343581  13.07.2015 09:40:00    990.27     17.71    291.68        12.88   
343582  13.07.2015 09:50:00    990.28     17.76    291.73        13.03   

        rh (%)  VPmax (mbar)  VPact (mbar)  VPdef (mbar)  sh (g/kg)  \
343578    76.9         19.22         14.78          4.44       9.34   
343579    76.3         19.48         14.86          4.62       9.39   
343580    74.0         19.94         14.76          5.19       9.32   
343581    73.3         20.30         14.88          5.42       9.40   
343582    73.8         20.36         15.03          5.33       9.49   

        H2OC (mmol/mol)  rho (g/m**3)  wv (m/s)  max. wv (m/s)  wd (deg)  
343578            14.92       1183.06   -9999.0        -9999.0     359.7  
343579            15.01       1181.97   -9999.0        -9999.0     359.8  
343580            14.90       1180.42   -9999.0        -9999.0     359.8  
343581            15.02       1179.26   -9999.0        -9999.0     359.8  
343582            15.17       1179.00   -9999.0        -9999.0     359.8

In [66]:
print(f'number of columns contain error value: {len(df[df["max. wv (m/s)"] < 0])}')
df[df["max. wv (m/s)"] < 0].head()

number of columns contain error value: 20


Date Time  p (mbar)  T (degC)  Tpot (K)  Tdew (degC)  \
343577  13.07.2015 09:00:00    990.52     16.98    290.93        12.94   
343578  13.07.2015 09:10:00    990.49     16.85    290.80        12.78   
343579  13.07.2015 09:20:00    990.32     17.06    291.03        12.86   
343580  13.07.2015 09:30:00    990.24     17.43    291.41        12.75   
343581  13.07.2015 09:40:00    990.27     17.71    291.68        12.88   

        rh (%)  VPmax (mbar)  VPact (mbar)  VPdef (mbar)  sh (g/kg)  \
343577    77.1         19.38         14.94          4.44       9.44   
343578    76.9         19.22         14.78          4.44       9.34   
343579    76.3         19.48         14.86          4.62       9.39   
343580    74.0         19.94         14.76          5.19       9.32   
343581    73.3         20.30         14.88          5.42       9.40   

        H2OC (mmol/mol)  rho (g/m**3)  wv (m/s)  max. wv (m/s)  wd (deg)  
343577            15.09       1182.50      6.16        -9999.0     289.6  
343578            14.92       1183.06  -9999.00        -9999.0     359.7  
343579            15.01       1181.97  -9999.00        -9999.0     359.8  
343580            14.90       1180.42  -9999.00        -9999.0     359.8  
343581            15.02       1179.26  -9999.00        -9999.0     359.8

In [67]:
df["Date Time"] = pd.to_datetime(df["Date Time"], format='%d.%m.%Y %H:%M:%S')
# df.iloc[1728]

In [68]:
df.corr()

Date Time  p (mbar)  T (degC)  Tpot (K)  Tdew (degC)  \
Date Time         1.000000  0.101275  0.110030  0.101240     0.128238   
p (mbar)          0.101275  1.000000 -0.045375 -0.124718    -0.066755   
T (degC)          0.110030 -0.045375  1.000000  0.996827     0.895708   
Tpot (K)          0.101240 -0.124718  0.996827  1.000000     0.894911   
Tdew (degC)       0.128238 -0.066755  0.895708  0.894911     1.000000   
rh (%)           -0.011241 -0.018352 -0.572416 -0.567127    -0.156615   
VPmax (mbar)      0.086787 -0.031546  0.951113  0.947293     0.799271   
VPact (mbar)      0.108469 -0.054370  0.867673  0.866205     0.968344   
VPdef (mbar)      0.044477 -0.003401  0.761744  0.756962     0.435752   
sh (g/kg)         0.106553 -0.069762  0.866755  0.866533     0.967599   
H2OC (mmol/mol)   0.106658 -0.069804  0.867177  0.866955     0.968044   
rho (g/m**3)     -0.081860  0.307640 -0.963410 -0.981345    -0.885232   
wv (m/s)         -0.007682 -0.005701 -0.004689 -0.004195    -0.008718   
max. wv (m/s)    -0.008152 -0.007760 -0.002871 -0.002224    -0.009091   
wd (deg)          0.010823 -0.063258  0.038732  0.043599     0.049877   

                   rh (%)  VPmax (mbar)  VPact (mbar)  VPdef (mbar)  \
Date Time       -0.011241      0.086787      0.108469      0.044477   
p (mbar)        -0.018352     -0.031546     -0.054370     -0.003401   
T (degC)        -0.572416      0.951113      0.867673      0.761744   
Tpot (K)        -0.567127      0.947293      0.866205      0.756962   
Tdew (degC)     -0.156615      0.799271      0.968344      0.435752   
rh (%)           1.000000     -0.615842     -0.151494     -0.843835   
VPmax (mbar)    -0.615842      1.000000      0.824865      0.875588   
VPact (mbar)    -0.151494      0.824865      1.000000      0.449154   
VPdef (mbar)    -0.843835      0.875588      0.449154      1.000000   
sh (g/kg)       -0.150841      0.824460      0.999851      0.448641   
H2OC (mmol/mol) -0.150969      0.824493      0.999856      0.448689   
rho (g/m**3)     0.514282     -0.901536     -0.850241     -0.698290   
wv (m/s)        -0.005020     -0.004018     -0.009600      0.001852   
max. wv (m/s)   -0.009921     -0.002213     -0.010316      0.005317   
wd (deg)        -0.015912     -0.009583      0.018418     -0.030881   

                 sh (g/kg)  H2OC (mmol/mol)  rho (g/m**3)  wv (m/s)  \
Date Time         0.106553         0.106658     -0.081860 -0.007682   
p (mbar)         -0.069762        -0.069804      0.307640 -0.005701   
T (degC)          0.866755         0.867177     -0.963410 -0.004689   
Tpot (K)          0.866533         0.866955     -0.981345 -0.004195   
Tdew (degC)       0.967599         0.968044     -0.885232 -0.008718   
rh (%)           -0.150841        -0.150969      0.514282 -0.005020   
VPmax (mbar)      0.824460         0.824493     -0.901536 -0.004018   
VPact (mbar)      0.999851         0.999856     -0.850241 -0.009600   
VPdef (mbar)      0.448641         0.448689     -0.698290  0.001852   
sh (g/kg)         1.000000         0.999997     -0.853325 -0.009479   
H2OC (mmol/mol)   0.999997         1.000000     -0.853769 -0.009477   
rho (g/m**3)     -0.853325        -0.853769      1.000000  0.003240   
wv (m/s)         -0.009479        -0.009477      0.003240  1.000000   
max. wv (m/s)    -0.010163        -0.010158      0.001086  0.948477   
wd (deg)          0.019376         0.019607     -0.058072 -0.015322   

                 max. wv (m/s)  wd (deg)  
Date Time            -0.008152  0.010823  
p (mbar)             -0.007760 -0.063258  
T (degC)             -0.002871  0.038732  
Tpot (K)             -0.002224  0.043599  
Tdew (degC)          -0.009091  0.049877  
rh (%)               -0.009921 -0.015912  
VPmax (mbar)         -0.002213 -0.009583  
VPact (mbar)         -0.010316  0.018418  
VPdef (mbar)          0.005317 -0.030881  
sh (g/kg)            -0.010163  0.019376  
H2OC (mmol/mol)      -0.010158  0.019607  
rho (g/m**3)          0.001086 -0.058072  
wv (m/s)       